### PR TITLE
React.PropTypes, React.createClass replaced for RN 0.49+

### LINKED ISSUE #30: Replace React.createClass with createReactClass (react 15.5+)
Chrome console output, on RN 0.46.2:
```
ExceptionsManager.js:71 Warning: React.createClass is no longer supported. Use a plain JavaScript class instead. If you're not yet ready to migrate, create-react-class is available on npm as a drop-in replacement. (https://fb.me/migrating-from-react-create-class)
reactConsoleErrorHandler | @ | ExceptionsManager.js:71
-- | -- | --
  | printWarning | @ | warning.js:36
  | warning | @ | warning.js:60
  | get | @ | react.development.js:2125
  | (anonymous) | @ | ToolTip.ios.js:24
  | loadModuleImplementation | @ | require.js:174
  | guardedLoadModule | @ | require.js:126
  | _require | @ | require.js:110
  | (anonymous) | @ | components.js:53
  | loadModuleImplementation | @ | require.js:174
  | guardedLoadModule | @ | require.js:126
  | _require | @ | require.js:110
  | (anonymous) | @ | index.ios.js:86
  | loadModuleImplementation | @ | require.js:174
  | guardedLoadModule | @ | require.js:119
  | _require | @ | require.js:110
  | (anonymous) | @ | redux-logger.js:1
  | executeApplicationScript | @ | debuggerWorker.js:40
  | (anonymous) | @ | debuggerWorker.js:65
```

A sample PR to fix: https://github.com/maxs15/react-native-modalbox/pull/148/files

### LINKED ISSUE #31: React.PropTypes is removed in RN 0.49
What used to be a warning: `Warning: PropTypes has been moved to a separate package` is now a removal, and redbox error in RN 0.49.3:
screenshot 2017-10-10 13 32 06

Need a fix like https://github.com/clauderic/react-native-highlight-words/pull/4/files

### PR DESCRIPTION
**Pull request by fungilation**
Fixes #30, #31

Redbox errors otherwise.

### CODE REVIEW

**Review by adlenafane — APPROVED** [review]

**Comment by adlenafane** [comment]
@chirag04 Hope you'll have some time to review this. It's a blocker with RN 0.49. Thanks

**Review by wwwdata — APPROVED** [review]

File: .vscode/bookmarks.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+	"onDidClearBookmarkEmitter": {},
+	"onDidClearAllBookmarksEmitter": {},
+	"onDidAddBookmarkEmitter": {},
+	"onDidRemoveBookmarkEmitter": {},
+	"onDidUpdateBookmarkEmitter": {},
```
Comment (chirag04) [inline .vscode/bookmarks.json:6]: can you remove this file?

File: yarn.lock [hunk 2]
```diff
@@ -0,0 +1,99 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
+
+
+asap@~2.0.3:
```
Comment (chirag04) [inline yarn.lock:5]: remove this file please?

**Comment by chirag04** [comment]
just 2 nits and we can ship this. ideally we switch everything to es6 classes but it's fine for now

File: yarn.lock (hunk 2 above)
Comment (fungilation) [reply yarn.lock:5]: Disagree about removing this one. I use yarn, it's gaining popularity, I think it's better than npm and it's a drop-in replacement. Doesn't hurt to have the file, can be ignored if not using.

**Comment by fungilation** [comment]
@chirag04 how about you make the call for what you want to merge. Merge this and then you can delete "offending" files as you wish.

**Comment by jeremyeaton89** [comment]
Can either this PR or https://github.com/chirag04/react-native-tooltip/pull/32 possibly get merged soon, please?

**Pull request closed without merging**